src/Penrose/Style.hs

```diff
@@ -1480,6 +1497,28 @@ translateStyProg varEnv subEnv subProg styProg labelMap styVals =
               in trace "translateStyProg: " $ Right transWithPlugins
         Left errors -> Left errors
 
+-- Style AST preprocessing:
+-- For any anonymous statement only (e.g. `encourage near(x.shape, y.shape)`), 
+-- replace it with a named statement (`local.<UNIQUE_ID> = encourage near(x.shape, y.shape)`)
+-- Note the UNIQUE_ID only needs to be unique within a block (since local will assign another ID that's globally-unique)
+-- Leave all other statements unchanged
+
+anonKeyword :: String
+anonKeyword = "ANON"
+
+nameAnonStatements :: StyProg -> StyProg
+nameAnonStatements p = map (\(h, b) -> (h, nameAnonBlock b)) p
```

Comment:
As long as it’s okay to duplicate names across blocks, this should be okay. You know more about that part than I do!

Reply:
Yeah, should be fine since all Substance ids are unique